tests/nixos/fetch-git/testsupport/gitea-repo.nix

```diff
@@ -55,7 +55,8 @@ in
         def create(self):
           # create ssh remote repo
           gitea.succeed(f"""
-            git init --bare -b main /root/{self.name}
+            git init --bare -b main /root/{self.name} \
+            && git config --global uploadpack.allowAnySHA1InWant true
```

Comment:
What is the purpose of `uploadpack.allowAnySHA1InWant` here; could you add a comment?

I assume it's not a problem for Nix users because this happens on the `gitea` host for some initialization purpose - not on a clone that is a normal client.

Reply:
I removed this line.
I thought we need this setting to enable shallow cloning via ssh by revision, but it seems it's not needed.